### PR TITLE
[Distributed] Run distributed graph server inside DGL

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
DGL should run distributed servers itself, instead of having users to invoke the servers. This is helpful for the launch script.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: examples/pytorch/graphsage/experimental/README.md [hunk 1]
```diff
@@ -28,15 +28,31 @@ will be able to access the partitioned data.
 
 We need to run a server on each machine. Before running the servers, we need to update `ip_config.txt` with the right IP addresses.
 
+On each of the machines, set the following environment variables.
+
+```bash
+export DGL_ROLE=server
```
Comment (aksnzhy) [inline examples/pytorch/graphsage/experimental/README.md:34]: We set `DGL_ROLE=serve`r here. When to set `DGL_ROLE=client`?

File: examples/pytorch/graphsage/experimental/README.md [hunk 2]
```diff
@@ -28,15 +28,31 @@ will be able to access the partitioned data.
...
+export DGL_IP_CONFIG=ip_config.txt
+export DGL_CONF_PATH=data/ogb-product.json
+export DGL_NUM_CLIENT=4
+```
+
 ```bash
 # run server on machine 0
-python3 train_dist.py --server --graph-name ogb-product --id 0 --num-client 4 --conf_path data/ogb-product.json --ip_config ip_config.txt
+export DGL_SERVER_ID=0
+python3 train_dist.py
```
Comment (aksnzhy) [inline examples/pytorch/graphsage/experimental/README.md:43]: Can we use `python3 DGL_SERVER_ID=0 train_dist.py` here? In such way we can launch many server processes in the same machine.

File: examples/pytorch/graphsage/experimental/README.md [hunk 3]
```diff
@@ -45,11 +61,11 @@ Pytorch distributed requires one of the trainer process to be the master. Here w
 
 ```bash
 # run client on machine 0
-python3 -m torch.distributed.launch --nproc_per_node=1 --nnodes=4 --node_rank=0 --master_addr="172.31.16.250" --master_port=1234 train_dist.py --graph-name ogb-product --ip_config ip_config.txt --num-epochs 3 --num-client 4 --batch-size 1000 --lr 0.1
+python3 -m torch.distributed.launch --nproc_per_node=1 --nnodes=4 --node_rank=0 --master_addr="172.31.16.250" --master_port=1234 train_dist.py --graph-name ogb-product --ip_config ip_config.txt --num-epochs 3 --batch-size 1000 --lr 0.1
```
Comment (aksnzhy) [inline examples/pytorch/graphsage/experimental/README.md:64]: `DGL_ROLE=client`?

Comment (aksnzhy) [reply examples/pytorch/graphsage/experimental/README.md:64]: I think we should remove `--graph-name` here.

File: examples/pytorch/graphsage/experimental/README.md (hunk 1 above)
Comment (zheng-da) [reply examples/pytorch/graphsage/experimental/README.md:34]: the default behavior is working as a client. we don't need to set it to be client. we could ask users to set it to client, but it doesn't matter.

File: examples/pytorch/graphsage/experimental/README.md (hunk 2 above)
Comment (zheng-da) [reply examples/pytorch/graphsage/experimental/README.md:43]: sure. this is temporary modification. once, you finish launch script, we don't need users to run any of the commands.

File: examples/pytorch/graphsage/experimental/README.md (hunk 3 above)
Comment (zheng-da) [reply examples/pytorch/graphsage/experimental/README.md:64]: the client needs to know the graph name. otherwise, it cannot create a DistGraph client.

**Pull request merged**